### PR TITLE
Fix highlight issues on manifest preview table 

### PR DESCRIPTION
**Pull request by rrchai**
7f221f8
- Fix bug that highlight was not properly displayed. For `R < 4.0`, the `strings` are by default converted to `factors` in `data.frame` and `factors` are not working properly in `formatStyle` in `datatable()`. Add `options(stringsAsFactors = FALSE)` in the global setting to make compatible for different R versions.
-  Blank values are not highlighted properly if they are wrong values. `blank` in filled manifest was read as `NA` using `readr`, but the `validateModelManifest` output the blank values as `''`. So the error values from backend (`''`) will not match the value in loaded manifest (`NA`). 

> This is not reproducible, so change `NA` to `""` after `readr` can simply solve the issue. We do not need  both frontend and backend output to something else <br>
<del>In addition, the `''` will not be considered as **not highlight** in `formatStyle` function. </del>

Or we may need backend support to let `validateModelManifest` read manifest from R variable instead of file path or output error blank values to `NA`. Otherwise, update front-end can still deal with this issue for now.

ee07b3c
- Improve help message and remove row index in preview table section

### CODE REVIEW

File: app.R [hunk 1]
```diff
@@ -19,7 +19,7 @@ reticulate::import("sys")
 
 source_python("synLoginFun.py")
 source_python("metadataModelFuns.py")
-
+options(stringsAsFactors = FALSE) # stringsAsFactors = TRUE by default for R < 4.0
```
Comment (sujaypatil96) [inline app.R:22]: Nit: Consider moving this to `.RProfile` during refactor.

**Review by sujaypatil96 — APPROVED** [review]

File: app.R [hunk 2]
```diff
@@ -457,9 +456,7 @@ schema_to_display_lookup <- data.frame(schema_name, display_name)
     template_type_df <- schema_to_display_lookup[match(input$template_type, schema_to_display_lookup$display_name), 1, drop = F ]
     template_type <- as.character(template_type_df$schema_name)
 
-    annotation_status <- metadata_model$validateModelManifest(input$file1$datapath, template_type) %>% 
-                          rapply(., function(x) gsub("\'\'", "\' \'", x), how = "replace") %>% # change '' to ' '
-                          rapply(., function(x) gsub("^$", " ", x), how = "replace") # change empty value to (one space) blank
+    annotation_status <- metadata_model$validateModelManifest(input$file1$datapath, template_type)
```
Comment (rrchai) [inline app.R]: Only change `"NA"` to `""` in #L420 can solve the issue, no need to change both to one space

**Comment by rrchai** [comment]
I notice the 'sss' cell in \<filename\> is not highlighted in the preview table (see screenshot). 

Screen Shot 2021-04-13 at 10 50 42 PM

I thought it was bug for highlighting function, then I notice it is not recognized as an invalid value from backend. So this PR is ready for review. @sujaypatil96 @milen-sage

**Comment by milen-sage** [comment]
Yes, filenames are not part of the controlled vocabulary in the data model schema, so we don't check during validation. On manifest submission, if a filename is not found on Synapse it is not included in the manifest stored on Synapse. Ideally we want to warn the user if this is the case with some of their files. Right now we don't warn them and the submission of the manifest proceeds w/o errors, which may be misleading. 

We discussed this a bit earlier: if for some reason the metadata submitter has a manifest that contains filenames that do not match any of the filenames found on Synapse, the resulting uploaded manifest to Synapse is empty - there is no warning or error displayed, which we want to handle better in the future.

I agree this PR is good to review/test and merge if there are no further requested changes @sujaypatil96

**Pull request merged**